### PR TITLE
ticket: Refactor "thought-storage-api.js"

### PR DESCRIPTION
**Pull request by SamueleA**
I've done the refactor task.
I moved the function createDirectory to google-drive-api.js. Not sure if there are more functions that should be moved.

A few things not mentioned in the ticket:
1. I delete functions not used anywhere else and that weren't doing anything

2. There's a commented line TODO talking about duplicated code. I didn't get to fixing that yet, so it's not part of this pull request.

### CODE REVIEW

**Comment by DarthVanger** [comment]
Well done!
Nice find with the unused functions!
Code changes look good.

>I moved the function createDirectory to google-drive-api.js. Not sure if there are more functions that should be moved.

Yes there are more functions, but we can move them in another pull request.

For example:
- `getFiles()`, because it is a function to get files from a folder on google drive. It has nothing to do with notes. https://github.com/DarthVanger/NeuralNotes/blob/63d15948f15b45e079e8e0c5a924a68dfeeca4ac/src/storage/thought-storage-api.js#L191
- `createFile()`, same story https://github.com/DarthVanger/NeuralNotes/blob/63d15948f15b45e079e8e0c5a924a68dfeeca4ac/src/storage/thought-storage-api.js#L298
- `createEmptyFile()`, same story https://github.com/DarthVanger/NeuralNotes/blob/63d15948f15b45e079e8e0c5a924a68dfeeca4ac/src/storage/thought-storage-api.js#L322
- `udpateFile()`, same story https://github.com/DarthVanger/NeuralNotes/blob/63d15948f15b45e079e8e0c5a924a68dfeeca4ac/src/storage/thought-storage-api.js#L347
- `getTextFileContents()`, same story https://github.com/DarthVanger/NeuralNotes/blob/63d15948f15b45e079e8e0c5a924a68dfeeca4ac/src/storage/thought-storage-api.js#L449

## Commit messages
Please change commit messages to follow the style & be more descriptive:
- `deleted unused functions`
  -> `Delete unused functions from thought-storage-api.js`
- `moved create directory to google drive api` 
  -> `Move a function from thought-storage-api.js to google-drive-api.js`
- `changes thoughts to note; cleaned comments;`
  -> multiline commit message (with 2nd line being empty):
  ```
  Rename "thought" to "note" in thought-storage-api.js
  
  Also delete commented code.
  ```
  Ideally cleaning comments should've been a separate commit. 

## To sum up
Please reword the commit messages, and I will merge this pull request.

Regarding moving more functions from `thought-storage-api.js` to `google-drive-api.js` - let's do it in a separate pull request.

**Comment by DarthVanger** [comment]
Oh and I've just noticed the `package-lock.json` included in this pull requesst.
It shouldn't a part of this pull request or any of the commits, please remove it from here.

**Comment by SamueleA** [comment]
The change to commit message are done.
I deleted package-lock.json
Additionally, I added the file to .gitignore so it's not accidentally pushed again.
